### PR TITLE
Plugin's Output - Added the Output Masker class

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

According to the ticket: https://issues.corp.rapid7.com/browse/PLGN-230 that would be great to implement such a credentials masking on the SDK side. Those changes allows to mask credentials found in the actions/triggers/tasks outputs. If the string contains the 80% percent of the credentials characters its getting masked. To change the threshold its possible from the constant called `DEFAULT_REPLACE_THRESHOLD`.

The for loop used in the function `mask_str_in_dict` (line 104 in `util.py` file) is used in case that the recursive method would lead into an infinite loop and raises the recursive Python error. 

The secrets content is masked with constant `********` stars.

### How Has This Been Tested?
This was tested locally on `python_3_script` and `example` plugins. Also there are some unittests written to test such a masking behaviour. 

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [X] I have updated the documentation accordingly (if changes are required).
- [X] I have tested the changes locally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ after cloning this repository.
 
 ## Changelog
 
+* 5.1.2 - Added the `OutputMasker` class
```
Comment (mrinehart-r7) [inline README.md:142]:
```suggestion
* 5.1.2 - Added the `OutputMasker` class
```

**Comment by mrinehart-r7** [comment]
@igorski-r7 just curious - have you profiled this at all to see how it scales with load? Maybe try running a plugin in server mode and then use a tool like Vegeta to load test it. I'm just concerned if this will have an impact on performance.

Other than that, I think this is a great idea.

**Review by ablakley-r7 — COMMENTED** [review]
Hey this all looks great and very cool idea and implementation. Have no issues but would agree with Mike on his points. 

In addition to the issue with masking output unintentionally I think the most likely case would be when outputting username. For example, an action that returns all users might have the requesting user masked.

Is it possible to remove username from the list of keys to check?

**Comment by igorski-r7** [comment]
> Hey this all looks great and very cool idea and implementation. Have no issues but would agree with Mike on his points.
> 
> In addition to the issue with masking output unintentionally I think the most likely case would be when outputting username. For example, an action that returns all users might have the requesting user masked.
> 
> Is it possible to remove username from the list of keys to check?

Yes. I think that would be a great idea for removing the `username` from that list, leaving only the secrets to be on a way for being masked.

**Comment by igorski-r7** [comment]
> 

Sure. I will do the load tests and provide its details here 😄 For the case that`part of a credential shows up in a string (by chance) but is not labeled in any way as a credential` well, I don't think we have the possibility to mask it anyway without having it in the connection part (I mean the knowledge of full credential).

**Comment by igorski-r7** [comment]
@mrinehart-r7, @ablakley-r7 Here are the load testing results done with `locust`.

### Python 3 Script plugin
- Before changes:
before

- After changes:
after

Tested with number of 10 fake users sending POST request to return the plugin such a huge response. We can notice that the timings are pretty similar.

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by dlaverty-r7 — DISMISSED** [review]

**Review by dlaverty-r7 — APPROVED** [review]

**Review by llaszuk-r7 — APPROVED** [review]

**Pull request merged**